### PR TITLE
Added wdio tests for button group

### PR DESCRIPTION
**Pull request by mhemesath**
### Summary
Added webdriver.io tests for button group.

### CODE REVIEW

File: packages/terra-button-group/tests/nightwatch/OnChangeButtonGroup.jsx [hunk 1]
```diff
@@ -16,9 +16,7 @@ class buttonGroupExample extends React.Component {
   render() {
     return (
       <div>
-        <div id="selected-index">
-          <h3>Selected Button: {this.state.selectedIndex}</h3>
-        </div>
+        <h3>Selected Button: <span id="selected-index">{this.state.selectedIndex}</span></h3>
```
Comment (mhemesath) [inline packages/terra-button-group/tests/nightwatch/OnChangeButtonGroup.jsx]: this makes the test a little easier to validate

File: packages/terra-button-group/tests/jest/ButtonGroup.test.jsx [hunk 2]
```diff
@@ -4,6 +4,11 @@ import ButtonGroup from '../../src/ButtonGroup';
 const button1 = <ButtonGroup.Button key="1" />;
 const button2 = <ButtonGroup.Button key="2" />;
 
+it('should render an empty component', () => {
```
Comment (mhemesath) [inline packages/terra-button-group/tests/jest/ButtonGroup.test.jsx]: I removed a test nightwatch was doing with an empty button group and moved it into jest as its more effective at determining if things don't need to render.

File: packages/terra-button-group/package.json [hunk 3]
```diff
@@ -45,6 +45,6 @@
     "props-table": "props-table ./src/ButtonGroup.jsx ./src/ButtonGroupButton.jsx --out-dir ./docs/props-table",
     "test": "npm run test:jest && npm run test:nightwatch",
```
Comment (jmsv6d) [inline packages/terra-button-group/package.json]: Should update this script to run `test:wdio`

Comment (bjankord) [reply packages/terra-button-group/package.json]: Yeah, lets get this updated, then I think this is read to merge down.

**Pull request merged**